Code/Tools/AssetProcessor/native/ui/BuilderData.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#if !defined(Q_MOC_RUN)
+#include <native/assetprocessor.h>
+#include <AZCore/std/containers/vector.h>
+#include <AZCore/std/containers/unordered_map.h>
+#include <AzCore/std/string/string.h>
+#include <AzCore/std/smart_ptr/shared_ptr.h>
+#include <QObject>
+#endif
+
+namespace AzToolsFramework
+{
+    namespace AssetDatabase
+    {
+        class AssetDatabaseConnection;
+    }
+} // namespace AzToolsFramework
+
+namespace AssetProcessor
+{
+    class BuilderDataItem;
+
+    class BuilderData : public QObject
```

Comment:
what is BuilderData tracking?

Reply:
`BuilderData` is a class that contains all jobs' metrics, categorized by builders. It is currently used by `BuilderInfoMetricsModel` as the source of data, and will also be used by the builder list model (the left part of the builder tab) in my next step, where it needs to show the metrics of each builder (but not each job). The data structures it contains originally stayed inside class `BuilderInfoMetricsModel`, and I decided to pull it out to share with the builder list model in the next step.

Reply:
Would you mind adding these comments to this class in this header?

Reply:
That makes sense!

Reply:
resolved